devtools/etdump/buffer_data_sink.h

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+#pragma once
+
+#include <executorch/devtools/etdump/data_sink_base.h>
+#include <executorch/runtime/core/exec_aten/exec_aten.h>
+#include <executorch/runtime/core/span.h>
+
+namespace executorch {
+namespace etdump {
+
+/**
+ * BufferDataSink is a concrete implementation of the DataSinkBase class,
+ * designed to store debug data in a pre-allocated, user-owned buffer. This
+ * class provides methods to write raw data and tensor data into the buffer,
+ * ensuring proper alignment and managing padding as needed. It is the standard
+ * DataSink used by ETDumpGen.
```

Comment:
If this fact changes, the comment will be incorrect. It's generally better to avoid comments that say who uses a particular type/function/etc.

```suggestion
 * ensuring proper alignment and managing padding as needed.
```